File: Arrays-Strings/IntegerToRoman.py
# class Solution(object):
#     def intToRoman(self, num):
#         """
#         :type num: int
#         :rtype: str
#         """
#         val = [
#             1000, 900, 500, 400,
#             100, 90, 50, 40,
#             10, 9, 5, 4,
#             1
#         ]
#         symb = [
#             "M", "CM", "D", "CD",
#             "C", "XC", "L", "XL",
#             "X", "IX", "V", "IV",
#             "I"
#         ]
        
#         roman = ""
#         i = 0
#         while num > 0:
#             count = num // val[i]
#             roman += symb[i] * count
#             num %= val[i]
#             i += 1
#         return roman

class Solution(object):
    def intToRoman(self, num):
        """
        :type num: int
        :rtype: str
        """
        roman = ""
        storeIntRoman = [[1000, "M"], [900, "CM"], [500, "D"],
                     [400, "CD"], [100, "C"], [90, "XC"],
                     [50, "L"], [40, "XL"], [10, "X"],
                     [9, "IX"], [5, "V"], [4, "IV"],
                     [1, "I"]]
        for i in range(len(storeIntRoman)):
            count = num // storeIntRoman[i][0]
            roman += storeIntRoman[i][1] * count
            num-= storeIntRoman[i][0] * count
        return roman

File: Arrays-Strings/IntegerToRoman.java
class Solution {
    public String intToRoman(int num) {
        StringBuilder roman = new StringBuilder();
        while (num > 0){
            if (num >= 1000){
                for (int i = 0; i < (num/1000); i++){
                    roman.append('M');
                }
                num = num - (num/1000) * 1000;
            }
            else if (num >= 900){
                roman.append("CM");
                num -= 900;
            }
            else if (num >= 500){
                for (int i = 0; i < (num/500); i++){
                    roman.append('D');
                }
                num = num - (num/500) * 500;
            }
            else if (num >= 400){
                roman.append("CD");
                num -= 400;
            }
            else if (num >= 100){
                for (int i = 0; i < (num/100); i++){
                    roman.append('C');
                }
                num = num - (num/100) * 100;
            }
            else if (num >= 90){
                roman.append("XC");
                num -= 90;
            }
            else if (num >= 50){
                for (int i = 0; i < (num/50); i++){
                    roman.append('L');
                }
                num = num - (num/50) * 50;
            }
            else if (num >= 40){
                roman.append("XL");
                num -= 40;
            }
            else if (num >= 10){
                for (int i = 0; i < (num/10); i++){
                    roman.append('X');
                }
                num = num - (num/10) * 10;
            }
            else if (num == 9){
                roman.append("IX");
                num -= 9;
            }
            else if (num >= 5){
                for (int i = 0; i < (num/5); i++){
                    roman.append('V');
                }
                num = num - (num/5) * 5;
            }
            else if (num == 4){
                roman.append("IV");
                num -= 4;
            }
            else{
                for (int i = 0; i < (num); i++){
                    roman.append('I');
                }
                num = num - num;
            }
        }
        return roman.toString();   
    }
}
